File: app/src/main/java/com/chad/baserecyclerviewadapterhelper/entity/Person.java
package com.chad.baserecyclerviewadapterhelper.entity;

import com.chad.library.adapter.base.entity.MultiItemEntity;

/**
 * Created by luoxw on 2016/8/10.
 */

public class Person implements MultiItemEntity {
    public Person(String name, int age) {
        this.age = age;
        this.name = name;
    }

    public Person(int age) {
        this.age = age;
    }

    public String name;
    public int age;

    @Override
    public int getItemType() {
        return 2;
    }
}

File: app/src/main/java/com/apache/fastandroid/bean/Person.kt
package com.apache.fastandroid.bean

import android.os.Parcelable
import com.chad.library.adapter.base.entity.MultiItemEntity
import kotlinx.android.parcel.Parcelize

/**
 * Created by Jerry on 2021/9/9.
 */
@Parcelize
data class Person(val name:String, val age:Int? ): MultiItemEntity, Parcelable {

    val children: MutableList<Person> = mutableListOf()

    constructor(name: String,age: Int, parent: Person) : this(name,age) {
        parent.children.add(this)
    }

    override val itemType: Int
        get() = 2

    @get:Synchronized
    @set:Synchronized
    var deletePolicy = 0
}